src/graphcut/normalized_cut.jl

```diff
@@ -0,0 +1,132 @@
+"""
+```
+labels                = normalized_cut(W, [num_cuts], [thres])
+```
+
+Performs recursive two-way normalized graph cut as described in this paper - "Normalized Cuts and Image Segmentation" by Shi and Malik
+
+Parameters:  
+-    W              = Weighted adjacency matrix for graph.
+-    thres          = Subgraphs aren't split if best normalized cut is above this threshold
+-    num_cuts       = Number of cuts performed to determine optimal cut
+
+# Example
+```julia
+using LightGraphs, SimpleWeightedGraphs
+g = SimpleWeightedGraph(6)
+add_edge!(g, 1, 2, 1)
+add_edge!(g, 2, 3, 1)
+add_edge!(g, 1, 3, 1)
+add_edge!(g, 4, 5, 1)
+add_edge!(g, 4, 6, 1)
+add_edge!(g, 5, 6, 1)
+add_edge!(g, 1, 6, 0.1)
+add_edge!(g, 3, 4, 0.2)
+
+labels = normalized_cut(g.weights, 0.1)
+```
+"""
+function ncut{T<:Real}(W::SparseMatrixCSC{T,Int}, thres::Real = 0.01 ,num_cuts::Int = 10)
+
+    #returns normalized cut score
+    function ncut_cost(mask)
+        cut_cost = 0
+        rows = rowvals(W)
+        vals = nonzeros(W)
+        for i = 1:n
+            for j in nzrange(W, i)
+                row = rows[j]
+                if mask[i] != mask[row]
+                    cut_cost += vals[j]/2
```

Comment:
You'll want to use a linter (like LanguageServer.jl's) to ensure your spacing is consistent.